subsys/bluetooth/common/dummy.c

```diff
@@ -59,3 +60,23 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_LOG_MODE_IMMEDIATE), "Immediate logging "
 	     "on selected backend(s) not "
 	     "supported with the software Link Layer");
 #endif
+
+#if defined(CONFIG_BT_CONN) && defined(CONFIG_BT_HCI_HOST)
+/* The host needs more ACL buffers than maximum ACL links. This is because of
+ * the way we re-assemble ACL packets into L2CAP PDUs.
+ *
+ * We keep around the first buffer (that comes from the driver) to do
+ * re-assembly into, and if all links are re-assembling, there will be no buffer
+ * available for the HCI driver to allocate from.
+ *
+ * Fixing it properly involves a re-design of the HCI driver interface.
+ */
+#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+BUILD_ASSERT(CONFIG_BT_BUF_ACL_RX_COUNT > CONFIG_BT_MAX_CONN);
+
+#else /* !CONFIG_BT_HCI_ACL_FLOW_CONTROL */
+BUILD_ASSERT(BT_BUF_RX_COUNT > CONFIG_BT_MAX_CONN);
```

Comment:
You can add messages to these `BUILD_ASSERT` with helpful instruction on how to fix the issue, e.g.

```suggestion
BUILD_ASSERT(CONFIG_BT_BUF_ACL_RX_COUNT > CONFIG_BT_MAX_CONN, 
	     "CONFIG_BT_BUF_ACL_RX_COUNT needs to be greater than CONFIG_BT_MAX_CONN");

#else /* !CONFIG_BT_HCI_ACL_FLOW_CONTROL */
BUILD_ASSERT(BT_BUF_RX_COUNT > CONFIG_BT_MAX_CONN, 
	     "BT_BUF_RX_COUNT needs to be greater than CONFIG_BT_MAX_CONN. "
	     "BT_BUF_RX_COUNT can be increased by increasing either CONFIG_BT_BUF_EVT_RX_COUNT, "
	     "CONFIG_BT_BUF_ACL_RX_COUNT or "
	     "CONFIG_BT_ISO_RX_BUF_COUNT (if CONFIG_BT_ISO is enabled));
```

Reply:
I mean in the build assert you get the exact line. Then you open the file to know what's up and the "migration guide" is in the comment right above.

Reply:
"Don't need a migration guide, just read the code" :D 

Keep in mind that users may not be familiar with the innards of Zephyr, and probably don't want to go through the code, even if it is open source, when their simple applications suddenly stop compiling.

Reply:
In practice they will still have to though...
I'll add your suggestion.

Reply:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/68428270/20e4657e-d566-47d1-ac3a-d9b5bbaa1744)

After reading it, I'm not sure I see the value of the first message

Reply:
Yeah, that's fair :)